src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -454,9 +454,14 @@ AndroidJavaObject CreateNativeConfig(Configuration config)
             }
 
             // set DiscardedClasses
-            if (config.DiscardClasses != null && config.DiscardClasses.Length > 0)
+            if (config.DiscardClasses != null && config.DiscardClasses.Count > 0)
             {
-                obj.Call("setDiscardClasses", GetAndroidRegexPatternSetFromArray(config.DiscardClasses));
+                var patternsAsStrings = new string[config.DiscardClasses.Count];
+                foreach (var pattern in config.DiscardClasses)
+                {
+                    patternsAsStrings[config.DiscardClasses.IndexOf(pattern)] = pattern.ToString();
+                }
+                obj.Call("setDiscardClasses", GetAndroidRegexPatternSetFromArray(patternsAsStrings));
```

Comment:
Can this throw an exception in the case where the C# regex pattern is not valid for Java?

Reply:
added handling 👍